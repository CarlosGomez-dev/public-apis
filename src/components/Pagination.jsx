import './Pagination.scss';

export const Pagination = ({ totalResults, currentPage, pageLimit, setCurrentPage }) => {
  const totalPages = Math.ceil(totalResults / pageLimit);
  const isFirstPage = currentPage === 1;
  const isLastPage = currentPage === totalPages;
  return (
    <nav className='pagination'>
      <p>{totalResults} results</p>
      <div className='button-container'>
        <button onClick={() => setCurrentPage(1)} disabled={isFirstPage}>
          &#60;&#60;
        </button>
        <button onClick={() => setCurrentPage(currentPage - 1)} disabled={isFirstPage}>
          Prev
        </button>
        <button onClick={() => setCurrentPage(currentPage + 1)} disabled={isLastPage}>
          Next
        </button>
        <button onClick={() => setCurrentPage(totalPages)} disabled={isLastPage}>
          &#62;&#62;
        </button>
      </div>
      {!!totalPages && (
        <p>
          Page {Math.min(currentPage, totalPages)} of {totalPages}
        </p>
      )}
    </nav>
  );
};
