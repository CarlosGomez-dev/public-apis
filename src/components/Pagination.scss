@import '../colors';

.pagination {
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  align-items: center;
  margin: 1rem auto;
  p {
    order: 1;
    text-align: center;
  }
}

.button-container {
  display: flex;
  margin: 0.5rem 0;
  button {
    cursor: pointer;
    border: 1px solid $border-color;
    border-radius: 5px;
    padding: 0.5rem 1rem;
    margin: 0 0.3rem;
    color: $font-color-inverse;
    background-color: $button-color;
    transition: background-color 250ms;
    &:hover {
      background-color: $button-color-hover;
    }
    &:disabled {
      cursor: default;
      background-color: $button-color-disabled;
      color: $button-font-color-disabled;
    }
  }
}

@media (min-width: 768px) {
  .pagination {
    flex-direction: row;
    p {
      order: unset;
    }
  }
}
