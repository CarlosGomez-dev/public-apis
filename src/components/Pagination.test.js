import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Pagination } from './Pagination';

const totalResults = 50;
const currentPage = 2;
const pageLimit = 20;
const setCurrentPage = jest.fn(value => value);
const paginationProps = { totalResults, currentPage, pageLimit, setCurrentPage };
const buttons = {
  first: '<<',
  previous: 'Prev',
  next: 'Next',
  last: '>>',
};
const paginationElements = [
  `${totalResults} results`,
  `Page ${currentPage}`,
  ...Object.values(buttons),
];

describe('Pagination', () => {
  it('matches snapshot', () => {
    const { container } = render(<Pagination {...paginationProps} />);
    expect(container).toMatchSnapshot();
  });
  it.each(paginationElements)('contains pagination information and buttons (%s)', element => {
    render(<Pagination {...paginationProps} />);
    screen.getByText(new RegExp(element, 'i'));
  });
  it('buttons allow changing pages', () => {
    render(<Pagination {...paginationProps} />);
    const firstButton = screen.getByText(buttons.first);
    const prevButton = screen.getByText(buttons.previous);
    const nextButton = screen.getByText(buttons.next);
    const lastButton = screen.getByText(buttons.last);
    userEvent.click(prevButton);
    userEvent.click(firstButton);
    expect(setCurrentPage).toHaveBeenCalledTimes(2);
    userEvent.click(nextButton);
    userEvent.click(lastButton);
    expect(setCurrentPage).toHaveBeenCalledTimes(4);
  });
});
