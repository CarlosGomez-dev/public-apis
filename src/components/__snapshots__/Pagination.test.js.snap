// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination matches snapshot 1`] = `
<div>
  <nav
    class="pagination"
  >
    <p>
      50
       results
    </p>
    <div
      class="button-container"
    >
      <button>
        &lt;&lt;
      </button>
      <button>
        Prev
      </button>
      <button>
        Next
      </button>
      <button>
        &gt;&gt;
      </button>
    </div>
    <p>
      Page 
      2
       of 
      3
    </p>
  </nav>
</div>
`;
